src/libraries/System.CodeDom/src/Microsoft/CSharp/CSharpCodeGenerator.cs

```diff
@@ -144,16 +145,26 @@ private string QuoteSnippetStringCStyle(string value)
                         b.Append(value[++i]);
                     }
 
-                    b.Append("\" +");
-                    b.Append(Environment.NewLine);
-                    b.Append(indentObj.IndentationString);
-                    b.Append('\"');
+                    if (i != value.Length - 1)
+                    {
+                        b.Append("\" +");
+                        b.Append(Environment.NewLine);
+                        b.Append(indentObj.IndentationString);
+                        b.Append('\"');
+                        isStringMultiline = true;
+                    }
                 }
                 ++i;
             }
 
             b.Append('\"');
 
+            if (isStringMultiline)
+            {
+                b.Insert(0, '(');
+                b.Append(')');
```

Comment:
@pedrobsaila I am not expert here and do not understand how wrapping in a bracket would help, could you explain or is there a doc/reference I could check?

Reply:
The bracket forces the next function expression to be evaluated on the whole string instead of just the last substring

Reply:
I took the easy road by doing this, I could :
* improve it by doing a sort of lookup so i add the bracket to a splitted string only if it is used as argument to the next function expression but the code does not pass context information to the process of code generation
* or just stop splitting strings

If you see better solutions, let me know

Reply:
I see, thank you for explanation, I was expecting we just stop splitting strings, but I guess that is better for styling. I have no better suggestion, the fix LGTM. 

I see @steveharter mentioned in the [issue](https://github.com/dotnet/runtime/issues/91060#issuecomment-1693807786) `string concat using + when it shouldn't be doing that`, not sure what else we could use other than `+`, but what he might have a better solution.

Reply:
I think the bracket approach is fine.